### PR TITLE
Migrate to tools directory

### PR DESCRIPTION
**Pull request by odsod**
- tools: migrate fully to tools dir
- tools: migrate to newer git-verify-nodiff
- tools: bump golangci-lint

### CODE REVIEW

File: pkg/socketcan/fileconn.go [hunk 1]
```diff
@@ -30,8 +30,7 @@ type fileConn struct {
 	ra net.Addr
 }
 
-// ensure net.Conn interface
-var _ net.Conn = &fileConn{}
+var _ = net.Conn(&fileConn{})
```
Comment (alethenorio) [inline pkg/socketcan/fileconn.go:33]: The previous form is generally seen as a more idiomatic type of assertion in Go. Any specific reason for the change?

File: cmd/cantool/main.go [hunk 2]
```diff
@@ -163,7 +163,7 @@ func genGo(inputFile string, outputFile string) error {
 	if err != nil {
 		return err
 	}
-	if err := ioutil.WriteFile(outputFile, output, 0644); err != nil {
+	if err := ioutil.WriteFile(outputFile, output, 0600); err != nil {
```
[the remark this answers is not in the document]
Comment (odsod) [reply cmd/cantool/main.go:166]: Linter wants it

File: pkg/socketcan/fileconn.go (hunk 1 above)
Comment (odsod) [reply pkg/socketcan/fileconn.go:32]: Interesting! I've been seeing this form more than the other form when reading various code, but I'll change back

Comment (alethenorio) [reply pkg/socketcan/fileconn.go]: The previous form is asserting type, the latter is type converting it. It seems type assertion was the intended idea here but I won't claim to understand the technical differences between them under the hood

File: pkg/dbc/parser.go [hunk 3]
```diff
@@ -293,11 +293,11 @@ func (p *Parser) float() float64 {
 	}
 	tok := p.nextToken()
 	if tok.typ != scanner.Int && tok.typ != scanner.Float {
-		p.fail(p.peekToken().pos, "expected int or float")
+		p.failf(p.peekToken().pos, "expected int or float")
 	}
 	f, err := strconv.ParseFloat(tok.txt, 64)
 	if err != nil {
-		p.fail(tok.pos, "invalid float")
+		p.failf(tok.pos, "invalid float")
```
Comment (ohlsont) [inline pkg/dbc/parser.go:300]: why failf and not fail on all these, there is no sign in the string?

Comment (odsod) [reply pkg/dbc/parser.go:300]: Yes, but we don't have a fail method, only `failf`, so using it for both cases.

**Comment by odsod** [comment]
Thanks @ohlsont!

**Pull request merged**